### ISSUE TITLE
Mismatch between copyright/license file listing and corpus files

### ISSUE THREAD

**Issue by mmartin9684-sil** [BibleNLP/bible-parallel-corpus#3]
The corpus folder contains 109 files, while hte copyright/license file listing contains 132 entries.

**Comment by DCNemesis** [comment]
I'm fairly confident the ones marked as cc-by in the copyright file are missing from the corpus (because there was no folder with that name), but that's only eight.

**Comment by davidbaines** [comment]
I've extracted some data from the copr.htm files and added that to the translations and licenses spreadsheet. 
It shows that there could be in the region of 1000 resources available to share and we have currently have 684 texts in the repo.
So we've better coverage of the license information and hope that there are more files we can identify to add soon.

**Comment by janetzki** [comment]
The [metadata file](https://github.com/BibleNLP/ebible-corpus/blob/main/metadata/Copyright%20and%20license%20information.xlsx) contains some bibles (e.g., a Hindi bible (hin2017)) under a http://creativecommons.org/licenses/by-sa/4.0/ license that are not part of the published corpus. Do you know why this is the case?

**Comment by davidbaines** [comment]
Thank you for this note. There are a few discrepancies between the metadata file and the texts in the corpus. This is 
largely due to human (my) error. Eventually we hope to have these created with code, and that should eliminate most of 
these mismatches.
At the moment though there are several manual steps to create the corpus and it's all too easy for mismatches to creep 
in.On 15/06/2022 08:24, Jonathan Janetzki wrote:

**Comment by janetzki** [comment]
Hi @davidbaines, thanks a lot for your answer and your effort in putting this awesome corpus together. How did you get each bible into a single file? If you have some code to create it, could you share it, please? This would help me to automatically create semantic domain dictionaries from bibles.

**Comment by davidbaines** [comment]
We use extract_corpora.py or bulk_extract_corpora.py from [SIL's NLP repo](https://github.com/sillsdev/silnlp/tree/master/silnlp/common/) They have a CLI to read USFM files from [eBible](https://ebible.org/) and extract them into the one verse per line format. 

There's no need for a GPU when doing the extraction.

**Comment by DCNemesis** [comment]
@janetzki to pass on further instructions that were once given to me ...

```
The extract_corpora script (under silnlp.common) is the one you want.

 

Create a subfolder under ‘Paratext’ for each project (i.e., “$SIL_NLP_DATA_PATH > Paratext > project”) and unzip the project files into that subfolder.  Then give that subfolder name as the command line option when you run extract_corpora:

 

                python -m silnlp.common.extract_corpora <projectA> [<projectB> …]

 

You can extract multiple projects in one extract run.

 

You will find the extract files in “$SIL_NLP_DATA_PATH > MT > scripture” (and the key terms files will be in “SIL_NLP_DATA_PATH > MT > terms”).
```

and

```
You can also use the "silnlp.common.bulk_extract_corpora" script to extract all projects in a folder. This script also does not require you to set up a specific directory structure. We use poetry to manage our dependencies, so you will need to install it first and run "poetry install" to create a virtual environment and install the dependencies. Once that is done, you can run the script like this:

poetry run python -m silnlp.common.bulk_extract_corpora --input <projects_folder> --output <extracted_corpora_folder> --error-log <error_log_text_file>

The error log will contain all projects that failed to extract.
```

(h/t @ddaspit @mmartin9684-sil )

**Comment by janetzki** [comment]
Thank you so much for the instructions, @davidbaines @DCNemesis. They have worked so that `extract_corpora.py` successfully has put some bibles in a text file.
